### PR TITLE
Make sure all data is sent.

### PR DESCRIPTION
**Pull request by jgosmann**
Am I the only one reading documentation?

[socket.send](https://docs.python.org/3/library/socket.html#socket.socket.send):
> Applications are responsible for checking that all data has been sent; if only some of the data was transmitted, the application needs to attempt delivery of the remaining data.

Addresses the problems experienced in #152.

### CODE REVIEW

**Comment by jgosmann** [comment]
This might have a minor performance impact. At some point (not before the summer school) we should probably introduce a mechanism to drop data if it's produce too quickly. But we definitely should not allow partial messages to be transmitted.

Another future performance optimization would be, to make the select call on all sockets, so that we can write pending data to whichever socket is ready first.

**Comment by tbekolay** [comment]
This LGTM but I still think we should leave sparkle plots out of the release. This still seems like it'll make things more robust, so probably worth merging ATM... are there any other places where we call `socket.send`?

**Comment by jgosmann** [comment]
There was one other place ...

**Comment by jgosmann** [comment]
ah no, there are more

**Comment by jgosmann** [comment]
I hope, that's all now.

File: nengo_viz/swi.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ def handle_ws_request(self, args, db):
         key = key.encode('ascii')
         resp_data = (HSHAKE_RESP %
                      base64.b64encode(hashlib.sha1(key).digest()).decode('ascii'))
-        client.socket.send(resp_data.encode('ascii'))
+        client.socket.sendall(resp_data.encode('ascii'))
```
Comment (jgosmann) [inline nengo_viz/swi.py]: socket is still in nonblocking mode here, so we can use `socket.sendall` instead of `ClientSocket._sendall`

**Comment by tbekolay** [comment]
OK. I'm going to merge this to get it in this initial release, as hopefully it will fix some subtle bugs...

**Comment by tbekolay** [comment]
This does have a performance impact for me; `2d.py` runs about 2.5x on master, about 1.8x on this branch. That's a simple model though, so I'm not sure if it's that important?

**Comment by jgosmann** [comment]
I don't know, right now it can randomly break in theory.

**Comment by tbekolay** [comment]
Fair enough! In it goes.

**Comment by jgosmann** [comment]
Ah just changed it

**Comment by tbekolay** [comment]
Ah whoops. Make another PR I guess

**Comment by tcstewar** [comment]
> Am I the only one reading documentation?

Argh!  Thank you again, and I apologize...  My only defence is that I wrote that code almost 10 years ago....  Back in the days that I used 2-space tabs and horrible horrible spacing....

**Pull request merged**